docs/adding_packages/dependencies.md

```diff
@@ -1,20 +1,181 @@
+        # Some magic setup
+        self.run(f"./configure.sh {opts_args}")
+```
+
+### Overriding the provided properties from the consumer
+
+> **Note**: This was adding in [Conan 1.55](https://github.com/conan-io/conan/pull/12609) to the generators... we need to
+> write docs for when that's available
+
+## Adherence to Build Service
+
+It's very rare we layout "rules", most often it's guidelines, however in order to ensure graph and the package generated are usable
+for consumer, we do impose some limits on Conan features to provide a smoother first taste to using Conan.
+
+> **Note**: These are very specific to the ConanCenter being the default remote and may not be relevant to your specifc use case.
+
+* [Version ranges](https://docs.conan.io/en/latest/versioning/version_ranges.html) are not allowed.
 * Specify explicit [RREV](https://docs.conan.io/en/latest/versioning/revisions.html) (recipe revision) of dependencies is not allowed.
-* Vendoring in library source code should be removed (best effort) to avoid potential ODR violations. If upstream takes care to rename
-  symbols, it may be acceptable.
 * Only ConanCenter recipes are allowed in `requires`/`requirements()` and `build_requires`/`build_requirements()`.
-* If a requirement is conditional, this condition must not depend on [build context](https://docs.conan.io/en/1.35/devtools/build_requires.html#build-and-host-contexts). Build requirements don't have this constraint.
-* Forcing options of dependencies inside a recipe should be avoided, except if it is mandatory for the library - in which case it must
-  be enforced through the `validate()` methods.
+* `python_requires` are not allowed.
+
+### Version Ranges
+
+Version ranges are a useful Conan feature, [documentation here](https://docs.conan.io/en/latest/versioning/version_ranges.html). However,
+in the context of ConanCenter they pose a few key challenges when being used generally to consume packages, most notably:
+
+* Non-Deterministic `package-id`: With version ranges the newest compatible package may yield a different `package_id` than the one built
+  and published by ConanCenter resulting in frustrating error "no binaries found". For more context
+  see [this excellent explanation](https://github.com/conan-io/conan-center-index/pull/8831#issuecomment-1024526780).
+
+* Build Reproducibility: If consumers try to download and build the recipe at a later time, it may resolve to a different package version
+  that may generate a different binary (that may or may not be compatible). In order to prevent these types of issues, we have decided to
+  only allow exact requirements versions. This is a complicated issue,
+  [check this thread](https://github.com/conan-io/conan-center-index/pull/9140#discussion_r795461547) for more information.
+
+## Handling "internal" dependencies
+
+Vendoring in library source code should be removed (best effort) to avoid potential ODR violations. If upstream takes care to rename
```

Comment:
```suggestion
Vendoring in library source code should be removed (best effort) to avoid potential [ODR violations](https://en.cppreference.com/w/cpp/language/definition). If upstream takes care to rename
```